strategies/categorized/src/test/java/org/arquillian/smart/testing/strategies/categorized/CategorizedTestsDetectorTest.java

```diff
@@ -167,7 +182,41 @@ public void should_return_classes_without_first_category_when_reversed_is_used()
             new CategorizedTestsDetector(config).selectTestsFromClasses(classesToProcess);
 
         // then
-        assertThat(SmartTesting.getClasses(new HashSet<>(testSelection)))
-            .containsExactlyInAnyOrder(ThirdCategorizedClass.class, NonCategorizedClass.class);
+        TestSelection testSelectionWithMethods =
+            getTestSelectionWithMethods(WithCategorizedMethodsClass.class, "testWithSecondCategory",
+                "testWithThirdCategory", "testWithoutCategory");
+        verifyContent(testSelection, testSelectionWithMethods, ThirdCategorizedClass.class, NonCategorizedClass.class);
+    }
+
+    private void verifyContent(Collection<TestSelection> actualTestSelections, TestSelection expectedWithMethodsSelection,
```

Comment:
Just a suggestion, isn't this also a good candidate for the custom assertions?

Reply:
I was thinking about it, unfortunately, this is a verification of a collection with elements of a specific type. Collections are quite tricky - in this case as you can just provide that the assertion is for a collection, but you cannot specify the type of the elements. As AssertJ provides an assertion for a collection already, so it could be quite confusing...
On the other hand, why not - it will be used also in the failed strategy, so it makes sense. So I'll create `TestSelectionCollectionAssert` for it.

Reply:
Done :-)